### ISSUE TITLE
Topic wise statistics retrieval

### ISSUE THREAD

**Issue by Gotham25** [void666/in-memory-queue#2]
Hey,

I had a look on your queue data-structure. It exactly satisfies what I need. Good job on your work on this. One more thing, Can I have topic wise statistics like, total messages in a topic and number of messages processed so far in a topic and total topics created so far ?.

I'm sure there's a way to do/retrieve this.

Let me know your thoughts on the same.

**Comment by void666** [comment]
Are you referring to runtime analysis? Like that of kafka? Can be written as wrapper APIs on top of the code. Open to PRs :)

**Comment by Gotham25** [comment]
Yep. Something like that. I created a PR for the same. Haven't added any tests yet as I'm having some issue in running the tests. Can you let me know how to setup the environment for testing ?.

**Comment by Gotham25** [comment]
Hey @void66, Any update on the above PR ?.

**Comment by void666** [comment]
I haven't had the time to test it out yet. Will push to next release. Thanks.
